ocaml/src/asd_client.ml

```diff
@@ -356,62 +356,67 @@ let make_client buffer_pool ~conn_info (lido:string option)  =
      closer () >>= fun () ->
      Lwt.fail exn)
 
-let with_client buffer_pool ~conn_info (lido:string option) f =
+let with_client ~conn_info (lido:string option) f =
   (* TODO: validation here? or elsewhere *)
   let () =
     match conn_info.Networking2.ips with
     | [] -> failwith "empty ips list for asd_client.with_client";
     | _ -> ()
   in
-  make_client buffer_pool ~conn_info lido >>= fun (client, closer) ->
+  make_client ~conn_info lido >>= fun (client, closer) ->
   Lwt.finalize
     (fun () -> f client)
     closer
 
 class asd_osd (asd_id : string) (asd : client) =
-  object(self :# Osd.osd)
-  method get_option prio (k:key) =
-    asd # multi_get ~prio [k] >>= fun vcos ->
-    let ho = List.hd_exn vcos in
-    let r =
-      match ho with
-      | None -> None
-      | Some (v,c) -> Some v
-    in
-    Lwt.return r
-
-  method get_exn prio (k:key) =
-    self # get_option prio k
-    >>= function
-    | None -> Lwt.fail (Failure (Printf.sprintf
-                                   "Could not find key %s on asd %S"
-                                   (Slice.get_string_unsafe k) asd_id))
-    | Some v -> Lwt.return v
-
-  method multi_get prio keys =
-    asd # multi_get ~prio keys >>= fun vcos ->
-    Lwt.return
-      (List.map
-         (Option.map fst)
-         vcos)
+object(self :# Osd.key_value_osd)
+
+  method kvs =
+    object
+      method get_option prio (k:key) =
+        asd # multi_get ~prio [k] >>= fun vcos ->
+        let ho = List.hd_exn vcos in
+        let r =
+          match ho with
+          | None -> None
+          | Some (v,c) -> Some v
+        in
+        Lwt.return r
+
+      method get_exn prio (k:key) =
+        asd # multi_get ~prio [ k; ]
+        >>= function
+        | [ None ] -> Lwt.fail (Failure (Printf.sprintf
+                                       "Could not find key %s on asd %S"
+                                       (Slice.get_string_unsafe k) asd_id))
+        | [ Some (v, _) ] -> Lwt.return v
+        | _ -> assert false
```

Comment:
isn't it better to use the above `get_option` method ? 
(also, doesn't the asd object support a 'get' method?)